### PR TITLE
feat(filters): new filters layout

### PR DESCRIPTION
**Pull request by joaogarin**
This pull request introduces a new layout for the filters inside a sidenav.

- [x]  Make sidenav for filters
- [x]  Tests
- [x]  get a better icon for filters
- [x]  Fix fab on mobile
- [x]  Double scrollbar
- [x]  Responsive sidenav
- [x]  Make sure sidenav dynamic config is server ready (proper checks before checking window)
- [x] Test hostlistener and window check
- [x] Test responsive behaviour

screen shot 2017-07-21 at 14 37 30

screen shot 2017-07-18 at 16 26 14

Marking as do not merge since there is still a minor bug in mobile where the fab is not fixed (not a huge deal..but maybe easy to fix) https://github.com/angular/material2/issues/998

### CODE REVIEW

**Comment by mrjmd** [comment]
I played around with this branch some, it seems to work fine. I guess I still have some concerns about this approach purely from a UX perspective... if the filters are part of what makes this application especially cool I find it odd that we're hiding them by default.

On mobile this would make sense but for desktop it seems like they weren't taking up that much real estate. Having the content update in the background while the main section of the page is dimmed diminishes the impact too. I don't know, I'm just thinking out loud.

**Comment by joaogarin** [comment]
Yeah I agree there might be some tweaks to it to make it better..one would be to add a proper submit button and hide the navbar at that point, also do that on the form submit event. But not on each input change.. Its quite nice to be able to tweak your filters without constant noise of the items refreshing also so I think there are advantages to it..

I don't love the idea of duplicate stuff for mobile and desktop all the time..so I would try and keep a solution that works cross device. Thats one of Material's mottos also..the one I kind of like the most.

Also I would definitely prioritise mobile on it..kind of the approach brought up by google inbox. Its a mobile layout, maybe not optimal for desktop, but mobile comes first.

But open to some suggestions and experimentation.

**Comment by joaogarin** [comment]
Dont merge this one yet. I will work on making the sidenav responsive. this way on desktop the filters are always visible as facets. related : https://github.com/angular/material2/issues/1130

File: src/app/components/recipes-and-filters/recipes-and-filters.component.ts [hunk 1]
```diff
@@ -33,4 +41,33 @@ export class RecipesAndFiltersComponent {
...
+      } else {
+        this.sidenav.open();
+      }
+    }
+  }
+
+  /**
+   * Listen to window resize.
+   */
+  @HostListener('window:resize', ['$event'])
```
Comment (joaogarin) [inline src/app/components/recipes-and-filters/recipes-and-filters.component.ts]: I am pretty sure this is supported with universal but I want to make sure..just asked for some info on some of the universal chats to be sure this is not going to break when we start using universal.

Comment (joaogarin) [inline src/app/components/recipes-and-filters/recipes-and-filters.component.ts]: related https://github.com/angular/angular/issues/15000
although its scroll, if one is supported the other should be too.

**Comment by joaogarin** [comment]
I made some changes to the PR. The idea now basically is on larger desktop devices the filters will be visible by default as facets, they can still be hidden if a user wants too (I think its ok to let the user hide them).

On mobile it stays as it was as a sidenav and hidden by default.

**Comment by joaogarin** [comment]
Since this is pushing PR's that pass the travis check (we should fix it will open an issue) you can test it on https://contenta-angular.firebaseapp.com

**Comment by mrjmd** [comment]
Fixed the firebase deploy issue finally, but I did look at this on firebase before I fixed it, I like the open by default for desktop behavior a lot more. @joaogarin Is this ready in your view? Should we remove the `do not merge` tag?

**Comment by joaogarin** [comment]
If this previous commit passes its ready. Added responsive testing so we can test mobile behaviour as well in the case something works different for mobile. I will remove the do not merge tag.

**Comment by joaogarin** [comment]
@mrjmd on my side this is ready. I have added some responsive settings for karma that should fix the missing codecov part altough it was not updated here. 

https://codecov.io/gh/contentacms/contenta_angular/pull/42/src/src/app/components/recipes-and-filters/recipes-and-filters.component.ts#L51

Right now it tests mobile and desktop versions..its good to have this setup cause this way any responsive behaviour inside the app can be properly tested.

**Pull request merged**